### PR TITLE
PR: Batch operations for lattice.

### PR DESCRIPTION
**Pull request by tuopuu**
Changing update_node to update_nodes. The code, tests, docstrings and uml diagrams should be all done in this commit.

### CODE REVIEW

File: simphony/cuds/lattice.py [hunk 1]
```diff
@@ -72,19 +72,21 @@ def get_node(self, index):
...
-    def update_node(self, lat_node):
-        """Update the corresponding lattice node (data copied).
+    def update_nodes(self, nodes):
+        """Update the corresponding lattice nodes (data copied).
 
         Parameters
         ----------
-        lat_node : reference to a LatticeNode object
-            data copied from the given node
+        nodes : an iterator to a LatticeNode objects
```
Comment (itziakos) [inline simphony/cuds/lattice.py]: nodes is an `iterable`

File: simphony/io/h5_lattice.py [hunk 2]
```diff
@@ -86,22 +86,23 @@ def get_node(self, index):
...
 
-    def update_node(self, node):
+    def update_nodes(self, nodes):
         """ Updates H5Lattice data for a LatticeNode
 
         Parameters
         ----------
-        node : LatticeNode
-            reference to LatticeNode object
+        nodes : iterator of LatticeNodes
```
Comment (itziakos) [inline simphony/io/h5_lattice.py]: see https://github.com/simphony/simphony-common/pull/178/files#r34018783

**Comment by itziakos** [comment]
I wonder if we should also have `get_nodes`

**Comment by tuopuu** [comment]
> I wonder if we should also have get_nodes

Probably yes. This of course should apply to other CUDS containers as well. We can delay this merge, if we want to add get_xxxs at the same time.

Thanks for the corrections, @itziakos. I'll re-push soon.

**Comment by roigcarlo** [comment]
@itziakos @tuopuu What would be the signature of get_xxxs? Wouldn't it be the same as iter_nodes()?

**Comment by SGGgarcia** [comment]
> @itziakos @tuopuu What would be the signature of get_xxxs? Wouldn't it be the same as iter_nodes()?

I was thinking the same.
Anyway, about having `get_xxxs`, if we are changing the API to work with multiple items, maybe we should also consider the `has_xxxs` ?

**Comment by nathanfranklin** [comment]
Would `get_xxxs` method just return a list of the entities with the provided uids?  Besides having all methods being plural like `add_xxxs` , what would the benefit to the user be?  If the user wanted to access multiple entities wouldn't he or she just use the `iter_xxxs()` method.

**Comment by itziakos** [comment]
> Would get_xxxs method just return a list of the entities with the provided uids? Besides having all methods being plural like add_xxxs , what would the benefit to the user be? If the user wanted to access multiple entities wouldn't he or she just use the iter_xxxs() method.

Yes, get_xxxs will return a list instead of an iterator. This is a small difference and in short means that 

```python
def get_items(uids):
      return [item for item in self.iter_items(uids)]
```

There are some functions that treat iterators different than lists (e.g. lists have a count). But you are correct that at this stage this change will be mainly for consistency.

**Comment by itziakos** [comment]
> Anyway, about having get_xxxs, if we are changing the API to work with multiple items, maybe we should also consider the has_xxxs ?

While this is a good point, I think that it is not a common pattern to have such an operation. Yet I am not against it if it is decided.

**Comment by roigcarlo** [comment]
Just a comment related with the UML images. Since they will have to be regenerated every time we make a modification in the API, and provided that at least #164 #174, #175, #176 will need to do such changes I would prefer to to generate them in a separated PR before the release with all the changes together rather than separately.
If not I think it will be kind of messy for the other branches once we begin to merge them.

**Comment by tuopuu** [comment]
> generate them in a separated PR

That's OK for me. I can remove them from this PR.

**Comment by itziakos** [comment]
> generate them in a separated PR

Yes, it makes sense

**Pull request merged**